### DISCUSSION TITLE
Any way to use parser diagnostic result in other task of orchestrator strategy?

### DISCUSSION THREAD

**Discussion by SteveWolligandt** [stevearc/overseer.nvim#331] [Q&A]
Hi!
I am trying to create sort of a batch processing using the orchestrator strategy:
* Task 1 executes some tests and prints out the directory of the results which gets parsed
* Task 2 checks out the nightly branch
* Task 3 recompiles everything
* Task 4 re-executes the failed tests from Task 1 that are stored in its results directory

Is there any way to use the parsed variable from task 1 in task 4?

**Comment by stevearc** [comment]
Not directly, but you could use [list_tasks](https://github.com/stevearc/overseer.nvim/blob/master/doc/reference.md#list_tasksopts) to find all tasks and you can look for tasks with the same `parent_id` as the currently running task to find the others in the orchestrator bundle. But you'll have to use other clues to find the exact task that you want. I'd recommend putting a value in the `meta` table.

**Reply by SteveWolligandt** [reply]
Hi @stevearc, thanks for your quick answer!
While not responding to your suggestion I experimented a bit more with components and overseer in general to get more familiar with it.
I thought I could add the clue for each of the orchestrator's subtasks via an individual component. Unfortunately, the subtasks of the orchestrator parent task do not have the components I specified. I tried something like this:

``` lua
overseer.register_template({
  name = "_batch-step-0",
  builder = function(_)
    return {
      cmd = {'git'},
      args = {'checkout', 'nightly'},
      components = {
        "default",
        "batch.task1",
      }
    }
  end,
  hide = true,
})

overseer.register_template({
  name = "_batch-step-1",
  builder = function(_)
    return {
      cmd = {'git'},
      args = {'checkout', 'nightly'}, -- this is just for testing
      components = {
        "default",
        "batch.task0",
      }
    }
  end,
  hide = true,
})

local batch_task = overseer.new_task({
  name = "Batch",
  strategy = {
    "orchestrator",
    tasks = {
      "_batch-step-0",              
      "_batch-step-1",
    },
    builder = function(_)
      return {
        components = {
          "default",
          "starccm.startestbatch.orchestrator",
        }
      }
    end,
  },
})
batch_task:start()
```
The list of components looks like this:
![image](https://github.com/user-attachments/assets/2dbc414b-b1d4-4653-99e0-7b36bc7e36d4)

How could I add the clue otherwise? Would you just use `list_tasks`, get the tasks with `task.parent_id == batch_task.id` and check for the subtask's names? This seems pretty inelegant to me.
